myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/NMExecutorCLGenImpl.java

```diff
@@ -0,0 +1,196 @@
+    NMPorts ports) {
+    this.cfg = cfg;
+    this.profile = profile;
+    this.ports = ports;
+  }
+
+  @Override
+  public String generateCommandLine() {
+    StringBuffer cmdLine = new StringBuffer();
+
+    generateEnvironment();
+    appendCgroupsCmds(cmdLine);
+    appendYarnHomeExport(cmdLine);
+    appendEnvForNM(cmdLine);
+    cmdLine.append(YARN_NM_CMD);
+    return cmdLine.toString();
+  }
+
+  protected void generateEnvironment() {
+    //yarnEnvironemnt configuration from yaml file 
+    Map<String, String> yarnEnvironmentMap = cfg.getYarnEnvironment();
+    if (yarnEnvironmentMap != null) {
+      environment.putAll(yarnEnvironmentMap);
+    }
+
+    String rmHostName = System.getProperty(KEY_YARN_RM_HOSTNAME);
+    if (rmHostName != null && !rmHostName.isEmpty()) {
+      addYarnNodemanagerOpt(KEY_YARN_RM_HOSTNAME, rmHostName);
+    }
+
+    if (cfg.getNodeManagerConfiguration().getCgroups().or(Boolean.FALSE)) {
+      addYarnNodemanagerOpt(KEY_YARN_NM_CONTAINER_EXECUTOR_CLASS,
+          VAL_YARN_NM_CONTAINER_EXECUTOR_CLASS);
+      addYarnNodemanagerOpt(KEY_YARN_NM_LCE_RH_CLASS, VAL_YARN_NM_LCE_RH_CLASS);
+
+        // TODO: Configure hierarchy
+        addYarnNodemanagerOpt(KEY_YARN_NM_LCE_CGROUPS_HIERARCHY,
+          VAL_YARN_NM_LCE_CGROUPS_HIERARCHY);
+        addYarnNodemanagerOpt(KEY_YARN_NM_LCE_CGROUPS_MOUNT, "true");
+        // TODO: Make it configurable
+        addYarnNodemanagerOpt(KEY_YARN_NM_LCE_CGROUPS_MOUNT_PATH,
+          VAL_YARN_NM_LCE_CGROUPS_MOUNT_PATH);
+        addYarnNodemanagerOpt(KEY_YARN_NM_LCE_GROUP, "root");
+        if (environment.containsKey("YARN_HOME")) {
+          addYarnNodemanagerOpt(KEY_YARN_HOME, environment.get("YARN_HOME"));
+        }
+    } else {
+        // Otherwise configure to use Default
+      addYarnNodemanagerOpt(KEY_YARN_NM_CONTAINER_EXECUTOR_CLASS,
+          DEFAULT_YARN_NM_CONTAINER_EXECUTOR_CLASS);
+    }
+    addYarnNodemanagerOpt(KEY_NM_RESOURCE_CPU_VCORES,
+      Integer.toString(profile.getCpus().intValue()));
+    addYarnNodemanagerOpt(KEY_NM_RESOURCE_MEM_MB,
+      Integer.toString(profile.getMemory().intValue()));
+    addYarnNodemanagerOpt(KEY_NM_ADDRESS, ALL_LOCAL_IPV4ADDR +
+      Long.valueOf(ports.getRpcPort()).toString());
+    addYarnNodemanagerOpt(KEY_NM_LOCALIZER_ADDRESS,
+      ALL_LOCAL_IPV4ADDR + Long.valueOf(ports.getLocalizerPort()).toString()); 
+    addYarnNodemanagerOpt(KEY_NM_WEBAPP_ADDRESS,
+      ALL_LOCAL_IPV4ADDR + Long.valueOf(ports.getWebAppHttpPort()).toString());
+    addYarnNodemanagerOpt(KEY_NM_SHUFFLE_PORT,
+      ALL_LOCAL_IPV4ADDR + Long.valueOf(ports.getShufflePort()).toString());
+  }
+
+  protected void appendEnvForNM(StringBuffer cmdLine) {
+    cmdLine.append(" env ");
+    for (Map.Entry<String, String> env : environment.entrySet()) {
+      cmdLine.append(env.getKey()).append("=").append("\"")
+          .append(env.getValue()).append("\" ");
+    }
+  }
```

Comment:
The CommandLineBuilder object has a setEnvironment method.  We probably should've used it previous. It may not be worth the refactor now, but I thought it's worth mentioning. The command line was long and now it's longer which is hard to debug.  Having a shorter command line and but calling env may be more pleasant.